fix: channel no longer being always allowed

Legend-Master opened the pull request:
<!--
Before submitting a PR, please read https://github.com/tauri-apps/tauri/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines

1. Give the PR a descriptive title.

  Examples of good title:
    - fix(windows): fix race condition in event loop
    - docs: update example for `App::show`
    - feat: add `Window::set_fullscreen`

  Examples of bad title:
    - fix #7123
    - update docs
    - fix bugs

2. If there is a related issue, reference it in the PR text, e.g. closes #123.
3. If this change requires a new version, then add a change file in `.changes` directory with the appropriate bump, see https://github.com/tauri-apps/tauri/blob/dev/.changes/README.md
4. Ensure that all your commits are signed https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits
5. Ensure `cargo test` and `cargo clippy` passes.
6. Propose your changes as a draft PR if your work is still in progress.
-->

We have introduced a small breaking change in #12890 that `Channel` will now follow ACL instead of always allowed, even though this is allowed in the default permission group, it's still a breaking change, let's delay this change to v3

~~Also changed the plugin name to `channel` to prepare for this change~~
Also commented a TODO about changing the plugin name to `channel` in v3

Legend-Master committed:
fix: channel no longer being always allowed

Legend-Master committed:
Prettier

amrbashir commented:
> Also changed the plugin name to `channel` to prepare for this change

That in itself is a breaking change

Legend-Master commented:
> That in itself is a breaking change

This is never exposed before and only used by our init scripts internally so I don't think it's considered a breaking change?

Legend-Master committed:
Don't change the plugin name just yet

Legend-Master commented:
Anyways, this can be done in v3 as well, reverted the plugin name change

lucasfernog commented:
we need to check if our optimization won't remove the channel fetch command though

Legend-Master committed:
Handle TAURI_CHANNEL as a special case in macro

amrbashir commented:
> > That in itself is a breaking change
> 
> This is never exposed before and only used by our init scripts internally so I don't think it's considered a breaking change?

Right, I was thinking the js package was using that but looks like it is not.

Legend-Master commented:
> we need to check if our optimization won't remove the channel fetch command though

I was thinking I removed the plugin mark so it will always get preserved but then I realized that it will just be considered as app's command, should be good to go now with 9fd5f391f26eb336050a7d306404ded5c31cb4e5, added back the special check

lucasfernog committed:
remove commented out code [skip ci]
